gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -473,61 +481,77 @@ def run(argv=None):
+  total_base_pairs = []
+  try:
     for i in range(num_shards):
-      table_suffix = sharding.get_output_table_suffix(i)
-      table_name = sample_info_table_schema_generator.compose_table_name(
-          known_args.output_table, table_suffix)
+      suffixes.append(sharding.get_output_table_suffix(i))
+      total_base_pairs.append(sharding.get_output_table_total_base_pairs(i))
       if not known_args.append:
+        table_name = sample_info_table_schema_generator.compose_table_name(
+            known_args.output_table, suffixes[i])
         pipeline_common.create_output_table(
-            table_name,
-            sharding.get_output_table_total_base_pairs(i),
-            schema_file)
-      _ = (variants[i] | 'VariantToBigQuery' + table_suffix >>
-           variant_to_bigquery.VariantToBigQuery(
-               table_name,
-               schema,
-               append=known_args.append,
-               allow_incompatible_records=known_args.allow_incompatible_records,
-               omit_empty_sample_calls=known_args.omit_empty_sample_calls,
-               num_bigquery_write_shards=known_args.num_bigquery_write_shards,
-               null_numeric_value_replacement=(
-                   known_args.null_numeric_value_replacement)))
-
-  if known_args.output_avro_path:
-    # TODO(bashir2): Add an integration test that outputs to Avro files and
-    # also imports to BigQuery. Then import those Avro outputs using the bq
-    # tool and verify that the two tables are identical.
-    for i in range(num_shards):
-      avro_path = sample_info_table_schema_generator.compose_table_name(
-          known_args.output_avro_path, sharding.get_output_table_suffix(i))
-      _ = (
-          variants[i] | 'VariantToAvro' >>
-          variant_to_avro.VariantToAvroFiles(
-              avro_path,
-              header_fields,
-              processed_variant_factory,
-              variant_merger=variant_merger,
-              allow_incompatible_records=known_args.allow_incompatible_records,
-              omit_empty_sample_calls=known_args.omit_empty_sample_calls,
-              null_numeric_value_replacement=(
-                  known_args.null_numeric_value_replacement))
-      )
-
-  result = pipeline.run()
-  result.wait_until_finish()
-
-  metrics_util.log_all_counters(result)
+            table_name, total_base_pairs[i], schema_file)
+        logging.info('Integer range partitioned table %s was created.',
+                     table_name)
+    load_avro = pipeline_common.LoadAvro(
+        avro_root_path, known_args.output_table, suffixes, total_base_pairs)
+    load_avro.start_loading()
+  except Exception as e:
+    logging.error('Something unexpected happened during the loading of AVRO '
+                  'files to BigQuery: %s', str(e))
+    logging.warning('Trying to revert as much as possible...')
+    if known_args.append:
+      logging.warning(
+          'Since tables were appended, we cannot revert added rows. You can '
```

Comment:
nit: s/'Since tables were appended, we cannot revert added rows.'/'Since tables were appended, added rows cannot be reverted.'